Tools/WebKitTestRunner/TestController.cpp

```diff
@@ -1952,47 +1952,53 @@ void TestController::didReceiveMessageFromInjectedBundle(WKStringRef messageName
+}
+
 void TestController::didReceiveAsyncMessageFromInjectedBundle(WKStringRef messageName, WKTypeRef messageBody, WKMessageListenerRef listener)
 {
-    auto completionHandler = [listener = retainWK(listener)] (WKTypeRef reply) {
+    CompletionHandler<void(WKTypeRef)> completionHandler = [listener = retainWK(listener)] (WKTypeRef reply) {
         WKMessageListenerSendReply(listener.get(), reply);
     };
 
     if (WKStringIsEqualToUTF8CString(messageName, "FlushConsoleLogs"))
         return completionHandler(nullptr);
 
     if (WKStringIsEqualToUTF8CString(messageName, "GetAllStorageAccessEntries"))
-        return TestController::singleton().getAllStorageAccessEntries(WTFMove(completionHandler));
+        return getAllStorageAccessEntries(WTFMove(completionHandler));
 
     if (WKStringIsEqualToUTF8CString(messageName, "GetAndClearReportedWindowProxyAccessDomains"))
-        return completionHandler(TestController::singleton().getAndClearReportedWindowProxyAccessDomains().get());
+        return completionHandler(getAndClearReportedWindowProxyAccessDomains().get());
 
     if (WKStringIsEqualToUTF8CString(messageName, "RemoveAllCookies"))
-        return TestController::singleton().removeAllCookies(WTFMove(completionHandler));
+        return removeAllCookies(WTFMove(completionHandler));
 
     if (WKStringIsEqualToUTF8CString(messageName, "TakeViewPortSnapshot"))
-        return completionHandler(TestController::singleton().takeViewPortSnapshot().get());
+        return completionHandler(takeViewPortSnapshot().get());
 
     if (WKStringIsEqualToUTF8CString(messageName, "SetStatisticsDebugMode"))
-        return TestController::singleton().setStatisticsDebugMode(booleanValue(messageBody), WTFMove(completionHandler));
+        return setStatisticsDebugMode(booleanValue(messageBody), WTFMove(completionHandler));
 
     if (WKStringIsEqualToUTF8CString(messageName, "SetStatisticsShouldBlockThirdPartyCookiesOnSitesWithoutUserInteraction"))
-        return TestController::singleton().setStatisticsShouldBlockThirdPartyCookies(booleanValue(messageBody), true, WTFMove(completionHandler));
+        return setStatisticsShouldBlockThirdPartyCookies(booleanValue(messageBody), true, WTFMove(completionHandler));
 
     if (WKStringIsEqualToUTF8CString(messageName, "SetStatisticsShouldBlockThirdPartyCookies"))
-        return TestController::singleton().setStatisticsShouldBlockThirdPartyCookies(booleanValue(messageBody), false, WTFMove(completionHandler));
+        return setStatisticsShouldBlockThirdPartyCookies(booleanValue(messageBody), false, WTFMove(completionHandler));
 
     if (WKStringIsEqualToUTF8CString(messageName, "SetStatisticsPrevalentResourceForDebugMode")) {
         WKStringRef hostName = stringValue(messageBody);
-        TestController::singleton().setStatisticsPrevalentResourceForDebugMode(hostName, WTFMove(completionHandler));
+        setStatisticsPrevalentResourceForDebugMode(hostName, WTFMove(completionHandler));
```

Comment:
You could do `return setStatisticsPrevalentResourceForDebugMode(hostName, WTFMove(completionHandler));` as the other places.

Reply:
I could.  I typically do that only if it makes the difference between having { } and not having { }, but it's a stylistic thing that doesn't matter much.